### PR TITLE
Address some issues found with Valgrind

### LINKED ISSUE #7478: tests: valgrind detected issues
Several issues reported by valgrind on tests

### tests/kernel/mbox/mbox_api/kernel.mailbox
```
starting test - test_mbox_block_get_invalid_pool
==21984== Thread 2:
==21984== Conditional jump or move depends on uninitialised value(s)
==21984==    at 0x8056B1F: mbox_message_match (mailbox.c:143)
==21984==    by 0x80579EF: k_mbox_get (mailbox.c:440)
==21984==    by 0x804AE23: tmbox_get (test_mbox_api.c:395)
==21984==    by 0x804B95F: tmbox_entry (test_mbox_api.c:538)
==21984==  Uninitialised value was created by a stack allocation
==21984==    at 0x804A4DE: tmbox_get (test_mbox_api.c:311)
==21984== 
==21984== Conditional jump or move depends on uninitialised value(s)
==21984==    at 0x80577DC: mbox_message_data_check (mailbox.c:411)
==21984==    by 0x8057A78: k_mbox_get (mailbox.c:447)
==21984==    by 0x804AE23: tmbox_get (test_mbox_api.c:395)
==21984==    by 0x804B95F: tmbox_entry (test_mbox_api.c:538)
==21984==  Uninitialised value was created by a stack allocation
==21984==    at 0x804A4DE: tmbox_get (test_mbox_api.c:311)
==21984== 
PASS - test_mbox_block_get_invalid_pool
```
### tests/kernel/critical/kernel.common
```
==23751== Conditional jump or move depends on uninitialised value(s)
==23751==    at 0x80491F9: atomic_and (atomic.h:250)
==23751==    by 0x80492F9: atomic_clear_bit (atomic.h:391)
==23751==    by 0x804936A: k_work_init (kernel.h:2253)
==23751==    by 0x8049799: critical_loop (main.c:87)
==23751==    by 0x8049A06: regression_task (main.c:152)
==23751==  Uninitialised value was created by a stack allocation
==23751==    at 0x8049721: critical_loop (main.c:80)
==23751== 
==23751== Conditional jump or move depends on uninitialised value(s)
==23751==    at 0x80491B7: atomic_or (atomic.h:208)
==23751==    by 0x804926F: atomic_test_and_set_bit (atomic.h:371)
==23751==    by 0x80493EF: k_work_submit_to_queue (kernel.h:2282)
==23751==    by 0x80497C9: critical_loop (main.c:88)
==23751==    by 0x8049A06: regression_task (main.c:152)
==23751==  Uninitialised value was created by a stack allocation
==23751==    at 0x8049721: critical_loop (main.c:80)
==23751== 
==23751== Thread 6:
==23751== Conditional jump or move depends on uninitialised value(s)
==23751==    at 0x8059856: atomic_and (atomic.h:250)
==23751==    by 0x80598D1: atomic_test_and_clear_bit (atomic.h:350)
==23751==    by 0x8059A73: work_q_main (work_q.c:37)
==23751==  Uninitialised value was created by a stack allocation
==23751==    at 0x8049721: critical_loop (main.c:80)
```
### drivers/crypto/test-micro
```
==24446== Use of uninitialised value of size 4
==24446==    at 0x804E55B: inv_sub_bytes (aes_decrypt.c:109)
==24446==    by 0x804E7EE: tc_aes_decrypt (aes_decrypt.c:148)
==24446==    by 0x804FBE5: tc_cbc_mode_decrypt (cbc_mode.c:106)
==24446==    by 0x8057107: do_cbc_decrypt (crypto_tc_shim.c:58)
==24446==    by 0x8049AEA: cipher_cbc_op (cipher.h:219)
==24446==    by 0x804A617: cbc_mode (main.c:179)
==24446==    by 0x804B675: zephyr_app_main (main.c:420)
==24446==    by 0x805872A: bg_thread_main (init.c:254)
==24446==  Uninitialised value was created by a stack allocation
==24446==    at 0x804A15F: cbc_mode (main.c:115)
==24446== 
[general] [INF] cbc_mode: CBC mode DECRYPT - Match
```

### PR DESCRIPTION
**Pull request by lpereira**
Please refer to #7478 for details.

I'm not really sure about the one fixing the mbox test; the warning is gone, but I don't know if a `struct k_mbox_msg` should be fully zeroed out prior to a `k_mbox_get()` call.

The k_work fix doesn't have a call to `atomic_clear_bit()` any longer; this shouldn't be an issue when setting its value to `0`, but I'd like a second opinion on this.

Fixes #7478

### CODE REVIEW

**Review by andyross — APPROVED** [review]
A few whines.  Nothing that should block merge unless you want to fix it.

File: samples/drivers/crypto/src/main.c [hunk 1]
```diff
@@ -212,11 +212,27 @@ u8_t ctr_ciphertext[64] = {
 void ctr_mode(void)
 {
 	struct device *dev;
-	struct cipher_ctx ini;
-	struct cipher_pkt  encrypt;
-	struct cipher_pkt decrypt;
 	u8_t encrypted[64] = {0};
```
[the remark this answers is not in the document]
Comment (lpereira) [reply samples/drivers/crypto/src/main.c]: The `{0}` is not necessary for Valgrind to stop whining; the initialization of these structs is.  Using these initializers will ensure that any member not initialized will be initialized to their respective zero value.

File: tests/kernel/mbox/mbox_api/src/test_mbox_api.c [hunk 2]
```diff
@@ -106,7 +106,7 @@ static void mbox_get_waiting_thread(void *thread_number, void *pmbox, void *p3)
 
 static void tmbox_put(struct k_mbox *pmbox)
 {
-	struct k_mbox_msg mmsg;
+	struct k_mbox_msg mmsg = {0};
 
 	memset(&mmsg, 0, sizeof(mmsg));
```
[the remark this answers is not in the document]
Comment (lpereira) [reply tests/kernel/mbox/mbox_api/src/test_mbox_api.c]: Yes, it was for symmetry.  I removed the call to memset and force-pushed.

**Comment by lpereira** [comment]
Pushed a new commit fixing the issue found by CI.  Unfortunately, there appears to be a warning/error with AddressSanitizer/Valgrind still.  I don't know what's causing the issue yet.

**Comment by lpereira** [comment]
Sent new revision updating the expected output and disabling log colors.

**Comment by lpereira** [comment]
Weirdly, only the `qemu_xtensa` board is showing failures.  All of them are timeouts.  Inspecting the outputs, they seem to be correct.

**Comment by andyross** [comment]
Xtensa is an oddball architecture in a lot of ways, including being the only USE_SWITCH architecture we have.  If libasan tries to walk stack frames, then it might be getting confused by the cross-stack calls we do for ISRs.   Also GCC's code generation also wastes stack space like crazy, so this is where stack overflows hit first.

**Comment by aescolar-ot** [comment]
> If libasan tries to walk stack frames

The address sanitizer is in a separate PR

**Comment by aescolar-ot** [comment]
>Weirdly, only the qemu_xtensa board is showing failures 

It seems to me that many, if not all boards, are timing out in the samples/drivers/crypto/test-mbedtls & micro

**Review by nashif — APPROVED** [review]

**Pull request merged**